src/Util.vala

```diff
@@ -166,23 +166,60 @@ namespace Tasks.Util {
     //--- X-Property ---//
 
 
-    public unowned ICal.Property? get_icalcomponent_x_property (ICal.Component ical_component, string x_property_name) {
+    public ICal.Property? get_icalcomponent_x_property (ICal.Component ical_component, string x_property_name) {
         return get_ecalpropertybag_x_property (new ECal.ComponentPropertyBag.from_component (ical_component, (property) => {
             return property.isa () == ICal.PropertyKind.X_PROPERTY;
         }), x_property_name);
     }
 
-    public unowned ICal.Property? get_ecalpropertybag_x_property (ECal.ComponentPropertyBag ecal_propertybag, string x_property_name) {
-        unowned ICal.Property? property = null;
-
+    public ICal.Property? get_ecalpropertybag_x_property (ECal.ComponentPropertyBag ecal_propertybag, string x_property_name) {
         var ecal_propertybag_count = ecal_propertybag.get_count ();
+
         for (int i = 0; i < ecal_propertybag_count; i++) {
-            property = ecal_propertybag.get (i);
+            unowned ICal.Property? property = ecal_propertybag.get (i);
             if (property.isa () == ICal.PropertyKind.X_PROPERTY && property.get_x_name () == x_property_name) {
-                break;
+                var x_property = new ICal.Property (ICal.PropertyKind.X_PROPERTY);
+                x_property.set_x_name (x_property_name);
+
+                if (property.get_x () != null) {
+                    x_property.set_x (property.get_x ().dup ());
+                }
+
+                return x_property;
+            }
+        }
+        return null;
+    }
+
+
+    //--- Sort ---//
+
+
+    public string get_ecalcomponent_sort (ECal.Component ecalcomponent) {
```

Comment:
Maybe this should more explicitly be something like `get_apple_sortorder` to explain why it's a string

Reply:
Changed to `public string? get_apple_sortorder_property_value`